### PR TITLE
Update Prometheus to v0.42.0 and Prometheus common to latest

### PR DESCRIPTION
**Pull request by rabenhorst**
As preparation for updating Thanos to Prometheus v0.42.0 we need to update the promql engine first. Additionally prometheus common is updated to latest version.

> Note: It was necessary to add a workaround for literal tests since the original Prometheus engine returns inconsistent results depending on the literal type (nil labels vs. empty labels).

### CODE REVIEW

File: engine/engine_test.go [hunk 1]
```diff
@@ -3262,3 +3264,15 @@ func roundValues(r *promql.Result) {
 		}
 	}
 }
+
+// nilEmptyLabels sets empty labelsets to nil to work around inconsistent
+// results from the old engine depending on the literal type (e.g. number vs. compare).
+func nilEmptyLabels(result *promql.Result) {
```
Comment (fpetkovski) [inline engine/engine_test.go:3270]:
```suggestion
func emptyLabelsToNil(result *promql.Result) {
```

**Review by fpetkovski — COMMENTED** [review]
Thanks, just one small suggestion.

File: engine/bench_test.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ func BenchmarkChunkDecoding(b *testing.B) {
 			ss := querier.Select(false, nil, matcher)
 			series := make([]chunkenc.Iterator, 0)
 			for ss.Next() {
-				series = append(series, ss.At().Iterator())
+				series = append(series, ss.At().Iterator(nil))
```
Comment (fpetkovski) [inline engine/bench_test.go:45]: Is there a way we can take advantage of reusing these iterators? Should we open an issue to investigate?

Comment (rabenhorst) [reply engine/bench_test.go:45]: I guess it's something worth looking into. I will create an issue and link details from Prom repo.
